.github/workflows/test.yml

```diff
@@ -108,67 +140,102 @@ jobs:
+          git config --global --add safe.directory '*'
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Install nextest
+        uses: taiki-e/install-action@nextest
+      - name: Download archive
+        uses: actions/download-artifact@v4
+        with:
+          name: nextest-archive-sqlite
+
+      - name: Test
+        run: |
+          cargo nextest run --archive-file nextest-archive-sqlite.tar.zst  --verbose --no-fail-fast \
           --workspace-remap $PWD
         timeout-minutes: 20
 
   test-integration:
-    needs: [build-test-artifacts, build-test-bins]
+    needs: build-test-bins
     strategy:
       matrix:
-        version: [02,03]
+        version: [02, 03]
         include:
           - version: 02
-            compose: "-f process-compose.yaml -D"
-            
+            compose: "-f process-compose.yaml"
+
           - version: 03
-            compose: "-f process-compose.yaml -f process-compose-mp.yml -D"
+            compose: "-f process-compose.yaml -f process-compose-mp.yml"
     runs-on: ubuntu-latest
     steps:
       - name: Install Nix
         uses: cachix/install-nix-action@V27
 
-      - name: Configure PATH
-        run: PATH="$PWD/target/debug:$PATH"
+      - uses: rui314/setup-mold@v1
 
       - uses: actions/checkout@v4
 
-      - uses: taiki-e/install-action@nextest
-
-      - name: Download archive
-        uses: actions/download-artifact@v4
-
-      - name: Move files
-        run: |
-          mv nextest-archive/* .
-          mkdir -p target/debug
-          mv test-binaries/* target/debug
-          chmod -c +x target/debug/*
+      - name: Enable Rust Caching
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-all-crates: true
+          cache-provider: buildjet
 
       - name: Install process-compose
         run: nix profile install nixpkgs#process-compose
+      
+      - name: Install nextest
+        uses: taiki-e/install-action@nextest
 
-      - name: Pull Docker Images
-        run: docker compose pull || docker-compose pull
-
-      - name: Run Demo-Native ${{matrix.version}}
-        run: bash -x scripts/demo-native ${{matrix.compose}}
+      - name: Download archive
+        uses: actions/download-artifact@v4
+        with:
+          name: test-binaries
+          path: target/debug
 
+      - name: chmod +x
+        run: chmod -c +x target/debug/*
+        
       - name: Test Integration
         env:
           NEXTEST_PROFILE: integration
           INTEGRATION_TEST_SEQUENCER_VERSION: ${{ matrix.version }}
         run: |
-          cargo nextest run --archive-file nextest-archive.tar.zst --verbose --no-fail-fast \
-          --workspace-remap $PWD $(if [ "${{ matrix.version }}" == "2" ]; then echo " smoke"; fi)
-        timeout-minutes: 40
+          cargo nextest run --no-run
```

Comment:
I think we should figure out of this was actually the problem. Since the tests in the `./tests` crate shouldn't change with the embedded-db feature. Any of the two nextest archives should be able to run the tests. And if the tests actually run I think it's very unlikely that the archive is the issue.

Reply:
yeah lets revert it